src/ui/components/SecondaryToolbox/index.tsx

```diff
@@ -158,18 +160,28 @@ export default function SecondaryToolbox() {
     ? toolboxLayout !== "ide"
     : toolboxLayout === "full";
 
+  // We definitely show these tabs if we have annotations.
+  // If we don't have annotations, we want to show the tabs anyway _if_ the recording
+  // is too long, in which case the panels will show a warning message describing why.
+  const shouldShowReactTab = hasReactAnnotations || !mightHaveRoutines;
+  const shouldShowReduxTab = hasReduxAnnotations || !mightHaveRoutines;
```

Comment:
This makes sense but also feels awkward. If a Vue or Angular user records something with Replay and sees a React component tab with this message, that's going to feel kind of janky to them.

Seems like we should ask BAC/RUN for a way to run some kind of scoped, much-smaller routine to determine if React exists at all and use that as a signal for this rather than just the length.